### PR TITLE
Itkdev: 5778: add stars to required fields

### PR DESCRIPTION
**Pull request by sinejespersen**
#### Link to issue

https://platform.dandigbib.org/issues/5778

#### Description

Add stars to required fields

#### Screenshot of the result

image

#### Checklist

- [x] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [x] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [x] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions

n/a

### CODE REVIEW

**Review by yepzdk — APPROVED** [review]

File: src/apps/patron-page/sections/PincodeSection.tsx [hunk 1]
```diff
@@ -85,7 +89,11 @@ const PincodeSection: FC<PincodeSectionProps> = ({
           type="password"
           onChange={(newPin) => setConfirmPincode(newPin)}
           value={confirmPincode}
-          label={t("patronPageConfirmPincodeLabelText")}
+          label={
+            required
+              ? `${t("patronPageConfirmPincodeLabelText")}*`
+              : t("patronPageConfirmPincodeLabelText")
```
Comment (spaceo) [inline src/apps/patron-page/sections/PincodeSection.tsx:95]: I think it should be the responsibility of the `TextInput` component to prefix label text with an asterisk if it is required.

File: src/components/contact-info-section/ContactInfoSection.tsx [hunk 2]
```diff
@@ -61,7 +61,7 @@ const ContactInfoSection: FC<ContactInfoSectionProps> = ({
         required
         onChange={(newEmail) => changePatron(newEmail, "emailAddress")}
         value={patron?.emailAddress}
-        label={t("patronContactEmailLabelText")}
+        label={`${t("patronContactEmailLabelText")}*`}
```
Comment (spaceo) [inline src/components/contact-info-section/ContactInfoSection.tsx:64]: I think it should be the responsibility of the `TextInput` component to prefix label text with an asterisk if it is required.

File: src/apps/patron-page/sections/PincodeSection.tsx [hunk 3]
```diff
@@ -73,7 +73,11 @@ const PincodeSection: FC<PincodeSectionProps> = ({
           inputmode="numeric"
           onChange={(newPin) => setPincode(newPin)}
           value={pincode}
-          label={t("patronPagePincodeLabelText")}
+          label={
+            required
+              ? `${t("patronPagePincodeLabelText")}*`
```
Comment (spaceo) [inline src/apps/patron-page/sections/PincodeSection.tsx:78]: I think it should be the responsibility of the `TextInput` component to prefix label text with an asterisk if it is required.

File: src/components/contact-info-section/ContactInfoSection.tsx (hunk 2 above)
Comment (sinejespersen) [reply src/components/contact-info-section/ContactInfoSection.tsx]: Good point

**Review by spaceo — APPROVED** [review]
LGTM 👍

**Pull request merged**